sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1120,14 +1133,10 @@ void captureSessionToken(RxDocumentServiceRequest request, RxDocumentServiceResp
         try {
             logger.debug("Creating a Document. collectionLink: [{}]", collectionLink);
 
-            Mono<RxDocumentServiceRequest> requestObs = getCreateDocumentRequest(collectionLink, document,
+            Mono<RxDocumentServiceRequest> requestObs = getCreateDocumentRequest(requestRetryPolicy, collectionLink, document,
                 options, disableAutomaticIdGeneration, OperationType.Create);
 
             Mono<RxDocumentServiceResponse> responseObservable = requestObs.flatMap(request -> {
-                if (requestRetryPolicy != null) {
-                    requestRetryPolicy.onBeforeSendRequest(request);
```

Comment:
fyi - I am not removing this onBeforeSendRequest , its just moved it early in the stack to getCreateDocumentRequest method , ensuring we are capturing meta data (collection calls) in retries

Reply:
the purpose of `getCreateDocumentRequest` is just to create the request. retry-policy interaction should happen outside. why are we moving this?